shim/src/evaluators.js

```diff
@@ -4,70 +4,65 @@
 import {
   apply,
   arrayJoin,
+  arrayFilter,
   arrayPop,
-  arrayPush,
   create,
   defineProperty,
   getOwnPropertyDescriptors,
   getOwnPropertyNames,
   getPrototypeOf,
   regexpMatch,
-  arrayForEach,
   setPrototypeOf,
   stringIncludes
 } from './commons';
 import { ScopeHandler } from './scopeHandler';
 import { rejectImportExpressions } from './block-imports';
 import { assert, throwTantrum } from './utilities';
 
+// admit many (but not all) legal variable names: starts with letter/_/$,
+// continues with letter/digit/_/$ . It will reject many legal names that
+// involve unicode characters. We use 'apply' rather than /../.match() in
+// case RegExp has been poisoned.
 const identifierPattern = /^[a-zA-Z_$][\w_$]*$/;
 
 // todo: think about how this interacts with endowments, check for conflicts
 // between the names being optimized and the ones added by endowments
 
 function getOptimizableGlobals(safeGlobal) {
-  const constants = [];
   const descs = getOwnPropertyDescriptors(safeGlobal);
 
-  arrayForEach(getOwnPropertyNames(descs), name => {
+  const constants = arrayFilter(getOwnPropertyNames(descs), name => {
```

Comment:
@warner Here is an update that might address what I think is yourc concern.

Reply:
yeah, that's marginally better.. I'd personally lean towards a list of `if x return false` clauses, but I'm happy with what you've got here too.